--- silo-core/contracts/incentives/SiloIncentivesController.sol	2025-01-29 17:14:24.738033308 +0100
+++ certora/patches/SiloIncentivesController.sol	2025-01-29 17:13:33.413371043 +0100
@@ -124,13 +124,13 @@
         virtual
         returns (bytes32 programId)
     {
-        string memory programName = Strings.toHexString(_tokenToDistribute);
-        programId = getProgramId(programName);
+        programId = keccak256(abi.encode(_tokenToDistribute)); // munge
+
 
         if (incentivesPrograms[programId].lastUpdateTimestamp == 0) {
             DistributionTypes.IncentivesProgramCreationInput memory _incentivesProgramInput;
 
-            _incentivesProgramInput.name = programName;
+            _incentivesProgramInput.name = "dummy"; // munge
             _incentivesProgramInput.rewardToken = _tokenToDistribute;
             _incentivesProgramInput.emissionPerSecond = 0;
             _incentivesProgramInput.distributionEnd = 0;
